common/script/index.coffee

```diff
@@ -72,13 +72,51 @@ api.daysSince = (yesterday, options = {}) ->
   Math.abs api.startOfDay(_.defaults {now:yesterday}, o).diff(api.startOfDay(_.defaults {now:o.now}, o), 'days')
 
 ###
-  Should the user do this taks on this date, given the task's repeat options and user.preferences.dayStart?
+  Should the user do this task on this date, given the task's repeat options and user.preferences.dayStart?
 ###
-api.shouldDo = (day, repeat, options={}) ->
-  return false unless repeat
+api.shouldDo = (day, dailyTask, options = {}) ->
+  return false unless dailyTask.type == 'daily' && dailyTask.repeat
+  # MIGRATION HACK, FIXME
+  if dailyTask.startDate == null
```

Comment:
Yup, a daily would always have a start date. I noticed that there's already a "creationDate" for each task, but it seems the creation date is meant more for recordkeeping of the actual creation date than something the user might change, hence my choice of adding this var. And oops! I'll remove the console.log!